e2e-test/src/test/java/org/alfresco/hxi_connector/e2e_test/CreateNodeE2eTest.java

```diff
@@ -0,0 +1,222 @@
+import org.apache.pdfbox.Loader;
+import org.apache.pdfbox.io.RandomAccessReadBuffer;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.text.PDFTextStripper;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.wiremock.integrations.testcontainers.WireMockContainer;
+
+import org.alfresco.hxi_connector.common.test.docker.repository.AlfrescoRepositoryContainer;
+import org.alfresco.hxi_connector.common.test.docker.util.DockerContainers;
+import org.alfresco.hxi_connector.common.test.util.RetryUtils;
+import org.alfresco.hxi_connector.e2e_test.util.client.AwsS3Client;
+import org.alfresco.hxi_connector.e2e_test.util.client.RepositoryNodesClient;
+import org.alfresco.hxi_connector.e2e_test.util.client.model.Node;
+import org.alfresco.hxi_connector.e2e_test.util.client.model.S3Object;
+
+@Slf4j
+@Testcontainers
+@SuppressWarnings({"PMD.FieldNamingConventions"})
+public class CreateNodeE2eTest
+{
+    private static final String BUCKET_NAME = "test-hxinsight-bucket";
+    private static final int INITIAL_DELAY_MS = 300;
+    private static final String PARENT_ID = "-my-";
+    private static final String DUMMY_CONTENT = "Dummy's file dummy content";
+
+    private static final Network network = Network.newNetwork();
+    @Container
+    private static final PostgreSQLContainer<?> postgres = DockerContainers.createPostgresContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> activemq = DockerContainers.createActiveMqContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> sfs = DockerContainers.createSfsContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> transformCore = DockerContainers.createTransformCoreAioContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> transformRouter = DockerContainers.createTransformRouterContainerWithin(network);
+    @Container
+    private static final AlfrescoRepositoryContainer repository = createRepositoryContainer();
+    @Container
+    private static final WireMockContainer hxInsightMock = DockerContainers.createWireMockContainerWithin(network)
+            .withFileSystemBind("src/test/resources/wiremock/hxinsight", "/home/wiremock", BindMode.READ_ONLY);
+    @Container
+    private static final LocalStackContainer awsMock = DockerContainers.createLocalStackContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> liveIngester = createLiveIngesterContainer();
```

Comment:
Probably not a change for this pr but maybe we could use dockercompose instead?

Reply:
This is worth considering 🤔 I'm not sure if run configuration for E2Es will be the same as this in `distribution` module